osquery/database/plugins/ephemeral.cpp

```diff
@@ -16,10 +18,36 @@ Status EphemeralDatabasePlugin::get(const std::string& domain,
                                     const std::string& key,
                                     std::string& value) const {
   if (db_.count(domain) > 0 && db_.at(domain).count(key) > 0) {
-    value = db_.at(domain).at(key);
+    try {
+      value = boost::get<std::string>(db_.at(domain).at(key));
+    } catch (const boost::bad_get& e) {
+      LOG(WARNING) << "Type error getting string value for (domain,key) : ("
+                   << key << "," << domain << ") " << e.what();
+      return Status(
+          1,
+          "EphemeralDatabasePlugin::get was requested incorrect type(string)");
+    }
+    return Status(0);
+  } else {
+    return Status(1, "Key or domain does not exist");
+  }
+}
+
+Status EphemeralDatabasePlugin::get(const std::string& domain,
```

Comment:
Looks like copy-paste from method for `std::string`. I'd suggest to use template here to avoid duplicity (with type check like
```
template<
  typename T,
  typename = typename std::enable_if<
    std::is_same<T, int>::value || std::is_same<T, str::string>::value
  >::type
>
```

Reply:
Thanks, Alex, it definitely looks nicer, however, I could not compile even simple template for some reasons. Would appriciate if you could make it work when its merged.

Reply:
Hmm, could you upload you version to the PR. Let's have a look.

Reply:
Oh I remember, it was problem of virtual function DatabasePlugin has virtual get, and I could not use template. But of course I can use template just for ephemeral.